lib/gst/player/gstplayer.c

```diff
@@ -2617,13 +2623,24 @@ gst_player_seek_internal_locked (GstPlayer * self)
   remove_tick_source (self);
   self->is_eos = FALSE;
 
-  ret =
-      gst_element_seek_simple (self->playbin, GST_FORMAT_TIME,
-      GST_SEEK_FLAG_FLUSH, position);
+  flags |= GST_SEEK_FLAG_FLUSH;
+  flags |= GST_SEEK_FLAG_ACCURATE;
 
-  if (!ret)
-    emit_error (self, g_error_new (GST_PLAYER_ERROR, GST_PLAYER_ERROR_FAILED,
+  if (self->trick_play)
+    flags |= GST_SEEK_FLAG_TRICKMODE;
+
+  s_event = gst_event_new_seek (self->rate, GST_FORMAT_TIME, flags,
+      GST_SEEK_TYPE_SET, position, GST_SEEK_TYPE_SET, GST_CLOCK_TIME_NONE);
```

Comment:
In theory we could also use the step events if the playback direction did not change, they are faster and cheaper. However if the rate changed too much, a seek event would still be useful (as elements can decide to only show keyframes then).